sdcm/logcollector.py

```diff
@@ -844,6 +844,33 @@ def save_kallsyms_map(node):
             LOGGER.error("Error occurred during collecting kallsyms on host: %s\n%s", node.name, details)
 
 
+def collect_diagnostic_data(node):
+    log_entities = [
+        # make names unique to avoid skipping
+        CommandLog(name=f'lsof_{datetime.datetime.now().strftime("%Y%m%d_%H%M%S")}', command='sudo lsof'),
+        CommandLog(name=f'netstat_{datetime.datetime.now().strftime("%Y%m%d_%H%M%S")}', command='sudo netstat -tanp'),
+    ]
+    collect_log_entities(node, log_entities)
+
+
+def collect_log_entities(node, log_entities: List[BaseLogEntity]):
+    """Collects diagnostics data from node - used during the test.
+
+    Log Entities should have unique names, otherwise won't be collected."""
+    LOGGER.info('Collecting diagnostics data from: %s', node.name)
+    if remote_node_dir := create_remote_storage_dir(node):
+        for log_entity in log_entities:
+            if os.path.exists(os.path.join(node.logdir, log_entity.name)):
+                LOGGER.debug("Diagnostic file '%s' already exists and not changed. Skipping collection.",
+                             log_entity.name)
+                continue
+            try:
+                log_entity.collect(node, node.logdir, remote_node_dir)
```

Comment:
the collect only bring it back to the test runner ? or upload to s3 ?

Reply:
to test runner, similar like kallsyms